compiler/natives/src/internal/unsafeheader/unsafeheader.go

```diff
@@ -0,0 +1,16 @@
+//go:build js
+// +build js
+
+package unsafeheader
+
+// Slice and String is Go's runtime representations which is different
+// from GopherJS's runtime representations. By purging these types,
+// it will prevent failures in JS where the code compiles fine but
+// expects there to be a constructor which doesn't exist when casting
+// from GopherJS's representation into Go's representation.
```

Comment:
I think I like this idea in theory, but I wonder if it may break user code that references those symbols but doesn't use them in a way that's unsupported. For example, an unsafe conversion `[]int -> unsafe.Slice -> []int` would still work.

Reply:
I felt it was safe to remove these since they are in an internal package and the comments on [the source code](https://cs.opensource.google/go/go/+/refs/tags/go1.20.14:src/internal/unsafeheader/unsafeheader.go) say,
> Slice is the runtime representation of a slice. It cannot be used safely or portably and its representation may change in a later release.

I actually based this off of [your commit for `TestWriteThroughHeader`](https://github.com/gopherjs/gopherjs/commit/4a391b4bbf145c9053c05016a25f65a492c92ff4) where you say,
> This test relies on assumptions about string and slice implementation that do not in fact hold true for GopherJS.

I figured that it would help me track down things that compile fine and may not be tested fully, but leverage those structures and therefore would break if used. By removing them it will keep them from compiling. It turns out, it was hardly used.

Reply:
Right, I missed this was an internal package. Then we don't risk breaking anyone besides ourselves, which we can deal with :) I think some public package defines similar structures, which threw me off.